features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/slowframes/SlowFramesListener.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.rum.internal.metric.slowframes
+
+import androidx.annotation.VisibleForTesting
+import androidx.metrics.performance.FrameData
+import com.datadog.android.rum.internal.domain.FrameMetricsData
+import com.datadog.android.rum.internal.domain.state.SlowFrameRecord
+import com.datadog.android.rum.internal.domain.state.ViewUIPerformanceData
+import com.datadog.android.rum.internal.vitals.FrameStateListener
+import com.datadog.tools.annotation.NoOpImplementation
+import kotlin.math.min
+
+@NoOpImplementation
+internal interface SlowFramesListener : FrameStateListener {
+    fun onViewCreated(viewId: String, startedTimestampNs: Long)
+    fun resolveReport(viewId: String): ViewUIPerformanceData
+}
+
+internal class DataDogSlowFramesListener(
+    @get:VisibleForTesting internal val maxSlowFramesAmount: Int = DEFAULT_SLOW_FRAME_RECORDS_MAX_AMOUNT,
+    @get:VisibleForTesting internal val frozenFrameThresholdNs: Long = DEFAULT_FROZEN_FRAME_THRESHOLD,
+    @get:VisibleForTesting internal val continuousSlowFrameThresholdNs: Long = DEFAULT_CONTINUOUS_SLOW_FRAME_THRESHOLD
```

Comment:
minor: I think we can avoid adding these annotations, they don't bring much value and were mostly designed for Java, where certain things had to be `public` for testing.